ble/UUID.h

```diff
@@ -42,7 +97,7 @@ class UUID {
      *         different service or characteristics on the BLE device.
      *
      * @param longUUID
-     *          The 128-bit (16-byte) UUID value, MSB first (big-endian).
+     *          The 128-bit (16-byte) UUID value, LSB first (little-endian).
```

Comment:
So the endian genie is already out of the bottle. The NRF51 service discovery is passing arrays in LSB to the UUID class.

https://github.com/ARMmbed/ble-nrf51822/blob/master/source/nRF5xServiceDiscovery.cpp#L253-L257